bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -935,46 +939,31 @@ protected void updateRunningAlarmProperties() {
         }
     }
 
-    protected void updateZoneInfo() {
+    protected boolean updateZoneInfo() {
         Map<String, String> result = service.invokeAction(this, "DeviceProperties", "GetZoneInfo", null);
-        Map<String, String> result2 = service.invokeAction(this, "DeviceProperties", "GetZoneAttributes", null);
-
-        result.putAll(result2);
-
         for (String variable : result.keySet()) {
             this.onValueReceived(variable, result.get(variable), "DeviceProperties");
         }
 
         Map<String, String> properties = editProperties();
-        boolean update = false;
-        if (StringUtils.isNotEmpty(this.stateMap.get("HardwareVersion"))
-                && !this.stateMap.get("HardwareVersion").equals(properties.get(Thing.PROPERTY_HARDWARE_VERSION))) {
-            update = true;
+        if (StringUtils.isNotEmpty(this.stateMap.get("HardwareVersion"))) {
             properties.put(Thing.PROPERTY_HARDWARE_VERSION, this.stateMap.get("HardwareVersion"));
         }
-        if (StringUtils.isNotEmpty(this.stateMap.get("DisplaySoftwareVersion")) && !this.stateMap
-                .get("DisplaySoftwareVersion").equals(properties.get(Thing.PROPERTY_FIRMWARE_VERSION))) {
-            update = true;
+        if (StringUtils.isNotEmpty(this.stateMap.get("DisplaySoftwareVersion"))) {
             properties.put(Thing.PROPERTY_FIRMWARE_VERSION, this.stateMap.get("DisplaySoftwareVersion"));
         }
-        if (StringUtils.isNotEmpty(this.stateMap.get("SerialNumber"))
-                && !this.stateMap.get("SerialNumber").equals(properties.get(Thing.PROPERTY_SERIAL_NUMBER))) {
-            update = true;
+        if (StringUtils.isNotEmpty(this.stateMap.get("SerialNumber"))) {
             properties.put(Thing.PROPERTY_SERIAL_NUMBER, this.stateMap.get("SerialNumber"));
         }
-        if (StringUtils.isNotEmpty(this.stateMap.get("MACAddress"))
-                && !this.stateMap.get("MACAddress").equals(properties.get(MAC_ADDRESS))) {
-            update = true;
+        if (StringUtils.isNotEmpty(this.stateMap.get("MACAddress"))) {
             properties.put(MAC_ADDRESS, this.stateMap.get("MACAddress"));
         }
-        if (StringUtils.isNotEmpty(this.stateMap.get("IPAddress"))
-                && !this.stateMap.get("IPAddress").equals(properties.get(IP_ADDRESS))) {
-            update = true;
+        if (StringUtils.isNotEmpty(this.stateMap.get("IPAddress"))) {
             properties.put(IP_ADDRESS, this.stateMap.get("IPAddress"));
         }
-        if (update) {
-            updateProperties(properties);
-        }
+        updateProperties(properties);
+
+        return !result.isEmpty();
```

Comment:
Why did you change the way this was implemented? It looks like it was intended to update properties if they haven't changed. Now you are updating the properties every time.

Reply:
I changed it because this was useless, the method `updateProperties` from the core framework is already checking if there is one change before updating the properties.